src/app/index.ts

```diff
@@ -289,12 +290,17 @@ module.exports = class extends yo {
       this.project = {
         folder: this.options.output || answerForName.name || this.options.name,
         name: this.options.name || answerForName.name,
-        host: this.options.host || answerForHost.host,
         projectType: _.toLower(this.options.projectType) || _.toLower(answerForProjectType.projectType),
         isManifestOnly: isManifestProject,
         isExcelFunctionsProject: isExcelFunctionsProject,
-        scriptType: answerForScriptType.scriptType ? answerForScriptType.scriptType : this.options.ts ? typescript : javascript
       };
+      this.project.host = answerForHost.host ? answerForHost.host 
+        : this.options.host ? this.options.host 
+        : jsonData.getHostTemplateNames(this.project.projectType)[0];
```

Comment:
If this is coming from outside of this method then we should check that it's not empty.